modules/data.land/R/soilgrids_texture_extract.R

```diff
@@ -0,0 +1,129 @@
+##' soilgrids_texture_extraction function
+##' A function to extract and save three types of soil texture data in parallel for a single or group of 
+##' lat/long locations based on user-defined site location from SoilGrids250m 
+##' version 2.0 : https://soilgrids.org
+##' @title soilgrids_texture_extraction
+##' @name soilgrids_texture_extraction
+##' 
+##' @param site_info A data frame of site info containing the BETYdb site ID, 
+##' site name, latitude, and longitude, e.g. 
+##' (site_id, site_name, lat, lon)
+##' @param outdir  Provide the path to store the texture data file
+##' @param verbose Provide progress feedback to the terminal? TRUE/FALSE
+##' @return a data frame containing the soil texture data with columns "Depth", "Quantile", "Siteid", and "Value"
+##' 
+##' @export
+##' @author Qianyu Li
+##' @importFrom magrittr %>%
+
+soilgrids_texture_extraction <- function(site_info, outdir=NULL, verbose=TRUE){
+  
+  # A function to extract and save one type of soil texture data 
+  download_and_extraction <- function(base_data_url, site_info, outdir) {
+    
+    if (future::supportsMulticore()) {
```

Comment:
Given the recommendations that you'll want to remove the parallel over the site locations, the parallel settings might also be removed from the script, if you don't mind.

Reply:
Fixed.